Add/blog new space (#98)
sheffieldhackspace/Public-Website: _posts/2025-08-01-new-space.md

<script type="module">
    import { Viewer } from '@photo-sphere-viewer/core';
    const baseURL = '{{site.baseurl}}/assets/blog/2025-08-01-new-space/2025-07-31/photospheres';
    [
      "a",
      "b1",
      "b2",
      "c",
      "d",
      "e",
      "f",
      "old",
    ].forEach(photoSphere => new Viewer({
        container: document.getElementById(`viewer-${photoSphere}`),
        panorama: `${baseURL}/room${photoSphere.toUpperCase()}.PHOTOSPHERE.webp`, 
      }));
</script>

Sheffield Hackspace has moved to a larger space within the same [Portland Works](https://www.portlandworks.co.uk/) building!

Members should find out more information via the [move wiki page](https://wiki.sheffieldhackspace.org.uk/members/move) or via Discord.

Prospective new members should read the [FAQ on the wiki](https://wiki.sheffieldhackspace.org.uk/).

While you can also come down to the space to have a look, let's have a digital explore! These pictures are from the 31st July 2025 (the first open day in the new space) — so it's tidier in reality than in the pics… we promise!

## Layout plans

Click/press on a room to jump to pictures of that room below!

<div class="layout-plan-container">

> {% include layout_proposal_v3.svg %}

Author: man how does this find the file... crazy jekyll

author: alifeee
layout: post
permalink: /new-space/
excerpt: "The hackspace has moved to a larger space in the same building. Take a look at some pictures!"
---
<style>
.layout-plan-container {
  width: 95vw;
  max-width: 95vw;
  overflow-x: scroll;
}
img#layout-plan {
  max-width: unset;
}
.gallery {
  line-height: 0;
  column-count: 2;
  column-gap: 0px;
}
.gallery > * {
  max-width: 100%;
  margin: 0;
}
p:has(img), .gallery {
  margin: 0;
}
.gallery img {
  width: 100% !important;
  height: auto !important;
  min-height: 2rem;
}
iframe, video {
  display: block;
  margin: 0.5rem;
  max-width: 100%;
  width: auto;
  height: auto;
}
.panorama {
  max-height: 80vh;
  max-width: 90vw;
  height: min(80vh, 50rem);
  width: min(90vw, 60rem);
}
.canvas {
  overflow: unset;
}
h2 {
  margin-top: 10rem;
  position: sticky;
  top: 0;
  background: white;
  z-index: 91;
  width: min(90vw, 60rem);
}
hr {
  margin-top: 2rem;
}
</style>
<link rel="stylesheet" href="{{site.baseurl}}/assets/blog/2025-08-01-new-space/photosphere.index.min.css" />
<script type="importmap">
    {
        "imports": {
            "three": "https://cdn.jsdelivr.net/npm/three/build/three.module.js",

>             "@photo-sphere-viewer/core": "https://cdn.jsdelivr.net/npm/@photo-sphere-viewer/core/index.module.js"

Reviewer: In this pr, you have both three.module.js and photo-sphere-viewer in the fs. Are we using jsdelivr or the filesystem? I would opt for removing the local fs files as the files from jsdelivr will be updated. On the other hand, it's scary to trust third parties to not turn against you one day. Your call.

Author: I usually prefer to serve from local filesystem, but importmap is freaky and imports a bunch of things, so I cba downloading them all

        panorama: '{{site.baseurl}}/assets/blog/2025-08-01-new-space/2025-07-31/roomC_20250731_171929674.PHOTOSPHERE.webp',
    });
    const viewer_d = new Viewer({
        container: document.querySelector('#viewer-d'),
        panorama: '{{site.baseurl}}/assets/blog/2025-08-01-new-space/2025-07-31/roomD_20250731_174747477.PHOTOSPHERE.webp',
    });
    const viewer_e = new Viewer({
        container: document.querySelector('#viewer-e'),
        panorama: '{{site.baseurl}}/assets/blog/2025-08-01-new-space/2025-07-31/roomE_20250731_172213307.PHOTOSPHERE.webp',
    });
    const viewer_f = new Viewer({
        container: document.querySelector('#viewer-f'),
        panorama: '{{site.baseurl}}/assets/blog/2025-08-01-new-space/2025-07-31/roomF_20250731_172808785.PHOTOSPHERE.webp',
    });
      const viewer_old = new Viewer({
        container: document.querySelector('#viewer-old'),
        panorama: '{{site.baseurl}}/assets/blog/2025-08-01-new-space/2025-07-31/old_20250727_105201756.PHOTOSPHERE.webp',
    });
</script>

<!-- resize image map to be same size as image -->
<script>
  function mapResizer(maps) {
    if (!maps) {maps = document.getElementsByTagName('map');}

>     for (const map of maps) {

Reviewer: NIT: This should probably be a `forEach` function

Author: feel free to push to branch to change it so long as it still works ;]

>         map.img = document.querySelectorAll(`[usemap="#${map.name}"]`)[0];

Author: feel free to push to branch to change it so long as it still works ;]

        map.areas = map.getElementsByTagName('area');

>         for (const area of map.areas) {

Reviewer: NIT: `foreach`

Reviewer: there's other but i won't go through it all now

Author: feel free to push to branch to change it so long as it still works ;]

            area.coordArr = area.coords.split(',');
        }
    }
    function resizeMaps() {
        for (const map of maps) {
            const scale = map.img.offsetWidth / (map.img.naturalWidth || map.img.width);
            for (const area of map.areas) {
                area.coords = area.coordArr.map(coord => Math.round(coord * scale)).join(',');
            }
        }
    }

>     window.addEventListener('resize', () => resizeMaps());

Reviewer: ```suggestion
    window.addEventListener('resize', resizeMaps);
```
I'm not sure you need to anonymise the function.

    resizeMaps();
}

> if (document.readyState == 'complete') {
>     mapResizer();
> } else {
>     window.addEventListener('load', () => mapResizer());
> }

Reviewer: I'm pretty sure that `document.readyState == 'complete'` will never be `true` because if the browser is reading this code, the page isn't loaded. The load event listener should be enough.

Also pretty sure the function doesn't need anonymisation.

```suggestion
window.addEventListener('load', () => mapResizer());
```

3. [Room B1 #](#room-b1)
   1. [Photosphere](#photosphere-1)
   2. [High-angle pictures](#high-angle-pictures-1)
4. [Room B2 #](#room-b2)
   1. [Photosphere](#photosphere-2)
   2. [High-angle pictures](#high-angle-pictures-2)
5. [Room C #](#room-c)
   1. [Photosphere](#photosphere-3)
   2. [High-angle pictures](#high-angle-pictures-3)
6. [Room D #](#room-d)
   1. [Photosphere](#photosphere-4)
   2. [High-angle pictures](#high-angle-pictures-4)
7. [Room E #](#room-e)
   1. [Photosphere](#photosphere-5)
   2. [High-angle pictures](#high-angle-pictures-5)
8. [Room F #](#room-f)
   1. [Photosphere](#photosphere-6)
   2. [High-angle pictures](#high-angle-pictures-6)
9. [Bonus](#bonus)
10. [The End](#the-end)

</details>

<figure class="gallery" markdown="1">
</figure>

-----

## Room A&nbsp;<a href="#room-a">#</a>

With long-term plans for metalwork, room A is currently the first entrance and storage.

### Photosphere

<div class="panorama" id="viewer-a"></div>

### High-angle pictures

![high-angle picture of room A]({{site.baseurl}}/assets/blog/2025-08-01-new-space/2025-07-31/roomA_20250731_173854809.webp)

-----

## Room B1&nbsp;<a href="#room-b1">#</a>

> Room B1 is eventually the "craft room", for sewing, vinyl cutting, leatherworking. It has room for future prospects like printing, other crafts, and perhaps larger machines like a more industrial sewing machine, interlocker, or embroidery machine. Currently – like room A – it's used a lot for storage.

Reviewer: ```suggestion
Room B1 will eventually become the "craft room", for sewing, vinyl cutting, leatherworking. It has room for future prospects like printing, other crafts, and perhaps larger machines like a more industrial sewing machine, interlocker, or embroidery machine. Currently – like room A – it's used a lot for storage.
```
